### ISSUE TITLE
0.59.x Discussion

### ISSUE THREAD

**Issue by grabbou** [react-native-community/react-native-releases#100]
Conversation on this thread are limited to 0.59.0 release's major issues and backport (cherry-pick) requests from commits that are already on master.

An example of a good such request is a bug fix for a serious issue that has been merged into master but did not make the 0.59 cut.

In other words, if you cannot point to a particular commit on master, then your request likely belongs as a new issue in http://github.com/facebook/react-native/issues.

If the commit you request to cherry pick is a complicated port, you will be asked to create the PR to the 0.59-stable branch yourself, in order to ensure that the process proceeds smoothly.

**Comment by gtebbutt** [comment]
Just missed the 0.59.0 cutoff, but could we cherry-pick https://github.com/facebook/react-native/commit/d834197746371b203bd7d7aaabdc2bc581acc867 please? It fixes the build on tvOS.

**Comment by grabbou** [comment]
Is tvOS build completely not working w/o this commit? Otherwise, I'd wait a couple of days to see if there are any other commits to include too.

**Comment by gtebbutt** [comment]
At least with the standard Xcode settings, it's a complete build failure (although the message describes it as a warning, there doesn't seem to be an option to ignore or override it). 

That said, Xcode also prompts to fix the file in-place and it's easy enough to add those two lines - and honestly, given the small percentage of us using it on tvOS anyway, I wouldn't personally be worried about another few days.

**Comment by ChrisEelmaa** [comment]
Can we get https://github.com/facebook/react-native/commit/fd954cda55b1f87d38bba7f42242163ecee78c35#diff-cf283d7cedacde5bd58c86627558a89c in?

we use TextInput with multiple lines and custom lineheight, the placeholder is not respecting lineheight currently having funky user experience.

**Comment by cjroth** [comment]
We should add a note to the documentation / website / blog that upgrading requires first updating RN in npm/yarn and then running `react-native upgrade --legacy`. Otherwise you'll get an error that react-native is already 0.59 and it won't upgrade.

**Comment by hramos** [comment]
@cjroth I just updated the blog post with some notes around the upgrade process (https://github.com/facebook/react-native-website/pull/828). Can you send a PR if you think that new section needs clarification?

**Comment by cjroth** [comment]
Nice. Blog makes sense, I'll try to find time to make a PR for the Docs.

I also just noticed that the metro debugger UI is 404ing (localhost:8081/debugger-ui/). Does it need to be specifically enabled now? If so we should document that as well.

**Comment by grabbou** [comment]
@cjroth we have just fixed that inside CLI. Reinstall `node_modules`'s and you are good to go.

**Comment by grabbou** [comment]
@ChrisEelmaa the commit you linked depends on another one (https://github.com/facebook/react-native/commit/65c014d19f933771a69c7cbccdbb744330cf13a3), which adds a new feature rather than fixing a bug. Would you be okay trying 0.60 in a couple of days instead?

**Comment by zhongwuzw** [comment]
Please cherry-pick this https://github.com/facebook/react-native/commit/4aa731ae76b0c764d6c3a9ee19cb516f54da3284 to fix x86_64 build.

**Comment by suther** [comment]
Would be great, if this would be fixed: https://github.com/facebook/react-native/issues/23342

**Comment by yurykorzun** [comment]
After updating to 0.59.1 I can no longer debug on Android, because generated source maps are off. Everything works well on iOS. Could it be that updated JSC messed it up?

**Comment by hramos** [comment]
@suther that's a link to a pull request opened before 0.59 was cut. We're only tracking cherry-pick requests for commits that are already in master.

**Comment by sunnylqm** [comment]
Please cheery pick https://github.com/facebook/react-native/commit/845189c17de621cc5aa373503220c1c12f649c3c

**Comment by dulmandakh** [comment]
Cherry picks:
- fix trackColor on Android - https://github.com/facebook/react-native/commit/d6ee448e15a25a7485482a4702aadb2e396445c7
- Do not use autofill methods on Android APIs older than Oreo - fixes crash - https://github.com/facebook/react-native/commit/d4aa1e7a52b51fa5d7fc9ded132b7b50170f2190

**Comment by mjmasn** [comment]
Just a heads up for anyone concerned there appears to be a fairly major issue with `<TouchableNativeFeedback />` in 0.59 if you use `onLongPress`.

https://github.com/facebook/react-native/issues/24001

**Comment by bdrobinson** [comment]
I don't think this is in any 0.59.x release yet but would be great to have if possible – fixing a bug where refresh control on ios gets stuck in a particular state and then stops being interactive: https://github.com/facebook/react-native/commit/95d399bc825c5471e08b83eff4b1b1b510e384a0

**Comment by adrianha** [comment]
Hi,
Please cherry pick https://github.com/facebook/react-native/commit/e38be82dfa8b49385b990629318f027de26500cf to solve broken single line text input when typing text reverts to default value

![TextInput behaviour](https://user-images.githubusercontent.com/5382429/54583704-984fa280-4a47-11e9-8c82-dc0bb73d9b7f.gif)

**Comment by dulmandakh** [comment]
cherry pick:
- use Conscrypt as security provider if available  - https://github.com/facebook/react-native/commit/75af15ede44135110e40de75a649d5b15430c590

**Comment by grabbou** [comment]
Thanks for the request. I am going to look into them and release a new version either today or tomorrow.

**Comment by danilobuerger** [comment]
Please cherry pick:
* Fixed regression in SectionList caused by #21577 not being able to scroll to top on android - https://github.com/facebook/react-native/commit/8fa116cc0e1cadbb6cf0734cfde0e0b8060f6b59

**Comment by jsamr** [comment]
Please cherry pick:

- Fix scrollview over bounds of content size [facebook/react-native@`9c1c5a7`](https://github.com/facebook/react-native/commit/9c1c5a7455d90ec837a9a6141c096de70b798e43)

**Comment by guhungry** [comment]
Please cherry pick

-  fix: Start Metro packager from project root https://github.com/facebook/react-native/commit/fe3aebf87b4123f8b16cdfcb9e2e774e6e0bf0b6

**Comment by RWOverdijk** [comment]
As referenced, https://github.com/facebook/react-native/pull/23585 would be great which fixes https://github.com/facebook/react-native/issues/24087

**Comment by grabbou** [comment]
Since we have decided to postpone 0.60 release over a week, I am going to do cherry-picking today and release a new version.

**Comment by sunnylqm** [comment]
Please cherry pick 
Fixed regression in SectionList caused by #21577 not being able to scroll to top on android
https://github.com/facebook/react-native/commit/8fa116cc0e1cadbb6cf0734cfde0e0b8060f6b59

**Comment by andyboythekid** [comment]
i can't seem to find which version this was merged into: https://github.com/facebook/react-native/pull/23446 would fix code-push image problems!

**Comment by grabbou** [comment]
@andyboythekid - it's not part of any of the releases yet. You can check the associated commit to this PR and you'll see it says "master", which means it's not part of any of the branches.

**Comment by grabbou** [comment]
Finished cherry-picking for 0.59.2. The build is now on the CircleCI. I'll check its status in a few hours and if it's all green, I will send it out.

**Comment by andyboythekid** [comment]
> @andyboythekid - it's not part of any of the releases yet. You can check the associated commit to this PR and you'll see it says "master", which means it's not part of any of the branches.

from what i can tell, this update is actually in 0.59.1, at least its working after upgrading for me! thanks!

**Comment by grabbou** [comment]
The release is ready to go. However, since it’s Friday, let’s wait till Monday morning to release it.

**Comment by grabbou** [comment]
0.59.2 is on CircleCI. Will hit `npm` in around 30 minutes.

**Comment by sunnylqm** [comment]
Please cherry pick 
https://github.com/facebook/react-native/commit/9db347fabca19c66f669faf4054c81cc3624be03

**Comment by kelset** [comment]
This would be good to cherry pick too https://github.com/facebook/react-native/commit/c13f5d48cfe3e7c0f6c6d0b745b50a089d6993ef

**Comment by radeno** [comment]
My two cents:

- https://github.com/facebook/react-native/commit/aad4dbbbfe937d1924e5359556979ab067198a58
- https://github.com/facebook/react-native/commit/395197dafe52355c8273f02e1952cdd4ba64a117 supplement to RN 0.59.2 which adds eslint
- https://github.com/facebook/react-native/commit/15619c22e57f73dfbed7bbe5fd6d9b3d2a8c9225 very handy
- https://github.com/facebook/react-native/commit/b3c74967ca6b20d7bda84c690ae3a99dfe255843
- https://github.com/facebook/react-native/commit/02e8e531ddfd86e9abf7ef47fbf30445afeb37cf
- https://github.com/facebook/react-native/commit/bbd98d5c46a37ebe23ce0424aff1d8c734c8b444
- https://github.com/facebook/react-native/commit/9db347fabca19c66f669faf4054c81cc3624be03
- https://github.com/facebook/react-native/commit/b943db418f4f0b9d0865642aaca3e1a2f1529663
- https://github.com/facebook/react-native/commit/56679ed359834c2177c8837d744cc7bf2ceb6b0a

**Comment by cjroth** [comment]
@grabbou re the debugger-ui not working - I have cleaned my yarn cache, removed all node modules, and then reinstalled everything. It's still happening.

**Comment by kelset** [comment]
hey everyone - I've cherry-picked most of the requests, and CI is green. Expect 0.59.3 later today or tomorrow.

**Comment by grabbou** [comment]
Can you outline @kelset what commits you have not cherry-picked? Would help to maintain this thread and also see what we can do with them (reject, ask authors to send a PR, see if they depend on another commit that can easily resolve the conflict).

**Comment by kelset** [comment]
Sure - sorry I totally forgot: I've cherry picked

* https://github.com/facebook/react-native/commit/b3c74967ca6b20d7bda84c690ae3a99dfe255843
* https://github.com/facebook/react-native/commit/02e8e531ddfd86e9abf7ef47fbf30445afeb37cf
* https://github.com/facebook/react-native/commit/aad4dbbbfe937d1924e5359556979ab067198a58
* https://github.com/facebook/react-native/commit/15619c22e57f73dfbed7bbe5fd6d9b3d2a8c9225
* https://github.com/facebook/react-native/commit/17415938c7180a95811db949122b8ad24a442866
* https://github.com/facebook/react-native/commit/9db347fabca19c66f669faf4054c81cc3624be03 (which required https://github.com/facebook/react-native/commit/581dc3ed212e405521f7229700f280d35ebfeff4)
* https://github.com/facebook/react-native/commit/b943db418f4f0b9d0865642aaca3e1a2f1529663
* https://github.com/facebook/react-native/commit/c13f5d48cfe3e7c0f6c6d0b745b50a089d6993ef

And I've added on top

* https://github.com/facebook/react-native/commit/ff66600224e78fec5d0e902f8a035b78ed31a961

Which should fix some issues devs are experiencing in Swift.

I couldn't cherry pick the others.

I'm waiting on CI to be green again after the last cherry pick, then I'll do the bump.

**Comment by Bardiamist** [comment]
Why https://github.com/facebook/react-native/commit/56679ed359834c2177c8837d744cc7bf2ceb6b0a not included?
I'm just trying to implement universal link in my project:)

**Comment by kelset** [comment]
@Bardiamist there was a conflict that looked fairly complex to solve. If you want to try and fix it for yourself (which is the procedure we follow in these scenarios) you can fork the main repo, switch to the 0.59 branch and try the cherry pick, and then submit a pr against that branch.

**Comment by Bardiamist** [comment]
@kelset I tried and no have conflitct on cherry pick. I think I should not create pr for this.

![Screenshot](https://i.imgur.com/HEaVlBl.png)

**Comment by kelset** [comment]
Uhm, weird. Ok let's try again that commit for 0.59.4

**Comment by kelset** [comment]
Anyway, now that 0.59.3 is out, let's move to a new issue to clean up the history & remove some noise -> https://github.com/react-native-community/react-native-releases/issues/109
